### DISCUSSION TITLE
Filtering a range of dates

### DISCUSSION THREAD

**Discussion by DaGaMa395** [sqlitebrowser/sqlitebrowser#3556] [Q&A]
I have a date/time column in this crazy-precise format 
2023-11-22 11:35:57.658020
I can't figure out how to filter a range, say all time stamps from 2024-01-03 to 2024-01-09.

**Comment by chrisjlocke** [comment]
I assume the field is text?  You could do it with SQL (but that creates a grid you can't directly edit), or a view with the field selected twice (to apply two filters) but again that creates a grid you can't edit (you can't edit views.. well, you can, but that's a rabbit hole you don't want to go down....) as only numeric fields allow a range operator (~).  
You could use regular expressions - '2024-01' and '01-03', and '01-04', etc... but that's a bit messy.
Another hacky alternative I use is to add a 'flag' field and set this flag using SQL, then in the data browse view, you can select just the 'flagged' records if you need to edit or filter further.
So short answer, not directly, but possibly if you need to.
Yell if you need more info. :)

**Reply by DaGaMa395** [reply]
Hi, thanks for your suggestions.
I'd already seen that I could get whole months or single days or hours by using expressions like your examples.
Changing the display format to "Julian day to date" datetime("clock") works correctly and displays "2023-11-22 11:35:57".
So the text in column "clock" is being recognised as a meaningful date and time value but filtering "2024-01-03 00:00:00~2024-01-10 00:00:00" gets nothing.
I have data every 5 seconds and need to extrapolate a week at a time. That's over 120000 lines...

**Reply by chrisjlocke** [reply]
If the field is text, you can simply go into the 'execute sql' tab and 'select blah blah where field1 between '2024-01-03 00:00' and '2024-01-09 23:59'

![image](https://github.com/sqlitebrowser/sqlitebrowser/assets/7872218/9ae9616f-f9c9-409a-9e1e-a1b6ff86f9e0)

What are you doing with the data once you extrapolate it .. are you just looking at it, or editing bits of it, etc, etc.

**Reply by chrisjlocke** [reply]
Another alternative is to add another column to the data, being the UNIX date value (which would be a float .. or otherwise known as a real ..) and being numeric, you could apply filters to it.  But that would be a lot of extra data, and if you're just getting stats, may not be needed.  So I'm just waffling now..... ;)
